src/bindable-property.js

```diff
@@ -58,17 +64,55 @@ export class BindableProperty {
   * @param descriptor The property descriptor for this property.
   */
   registerWith(target: Function, behavior: HtmlBehaviorResource, descriptor?: Object): void {
+    let { reflect } = this;
     behavior.properties.push(this);
     behavior.attributes[this.attribute] = this;
     this.owner = behavior;
 
+    if (reflect) {
+      behavior._registerReflection(this.name, typeof reflect === 'function' ? reflect : propToAttr);
+      this._configureReflection(target);
+    }
+
     if (descriptor) {
       this.descriptor = descriptor;
       return this._configureDescriptor(descriptor);
     }
 
     return undefined;
   }
+  
+  _configureReflection(target) {
+    if (target.__reflectionConfigured__) return;
+
+    let method = 'propertyChanged';
+    let onChanged = target.prototype[method];
+    let hasHandler = !!onChanged;
+    
+    let alteredHandler;
+    if (hasHandler) { // avoid ternary to make it consisten with the rest ?
+      alteredHandler = function propertyChanged(name, newVal, oldVal) {
+        onChanged.call(this, name, newVal, oldVal);
+        let { __element__, __reflections__ } = this.__observers__;
+        if (!__element__) return;
+        __reflections__[name].call(this, __element__, name, newVal, oldVal)
+      }
+    } else {
+      alteredHandler = function propertyChanged(name, newVal, oldVal) {
+        let { __element__, __reflections__ } = this.__observers__;
+        if (!__element__) return;
+        __reflections__[name].call(this, __element__, name, newVal, oldVal);
+      }
+    }
+  
+    if (!Reflect.defineProperty(target.prototype, method, {
+      configurable: true,
+      value: alteredHandler
+    })) {
+      throw new Error(`Cannot setup property reflection on <${this.name}/> for ${target.name}`);
```

Comment:
If we go for `throw` on failure, why don't use `Object.defineProperty`. Unlike `Reflect` it exists on older browsers, does the same thing, and throws when it fails.

Reply:
This is because my lack of knowledge about compiler result. I didn't know if `'use strict'` will always be there so i tried to do something.